### PR TITLE
Add checks to allow the bot owner to transfer/edit/delete tags.

### LINKED ISSUE #3: Add optional owner check to tag commands
In addition to tag owners and server admins being able to use tag commands such as `/tags edit`, `/tags delete`, and `/tags transfer`, the bot/instance owner should also be able to run the commands.

This could be accomplished through the `discord.ext.commands.Bot.is_owner` function.

### PR DESCRIPTION
**Pull request by bsrecker**
Included a check to allow the bot instance owner to edit/transfer/delete.

Additionally, moved this check to its own dedicated function to cut-down on duplicated code.

This closes #3.

### CODE REVIEW

File: leaf/extensions/tags.py [hunk 1]
```diff
@@ -14,6 +14,11 @@ class TagsCog(commands.GroupCog, name="Tags", group_name="tags"):
     def __init__(self, bot: LeafBot) -> None:
         self.bot = bot
 
+    async def check_perms(self, tag_record,  interaction: discord.Interaction) -> bool:
```
Comment (mudkipdev) [inline leaf/extensions/tags.py]: This should be renamed to check_permissions, add a type hint for tag_record, as well as removing the double space between the comma and the word "interaction".

File: leaf/extensions/tags.py [hunk 2]
```diff
@@ -14,6 +14,11 @@ class TagsCog(commands.GroupCog, name="Tags", group_name="tags"):
...
+        return tag_record["owner_id"] == interaction.user.id \
```
Comment (mudkipdev) [inline leaf/extensions/tags.py]: Would using paranthesis be cleaner here?
```py
return (
    tag_record["owner_id"] == interaction.user.id
    or ...
    or ...
)
```

File: leaf/extensions/tags.py [hunk 3]
```diff
@@ -188,8 +193,7 @@ async def edit_tag(self, interaction: discord.Interaction, tag: str) -> None:
                 return
 
             if (
-                tag_record["owner_id"] == interaction.user.id
-                or interaction.user.guild_permissions.manage_guild
+                await self.check_perms(tag_record, interaction)
```
Comment (mudkipdev) [inline leaf/extensions/tags.py]: Why new line here?

Comment (mudkipdev) [reply leaf/extensions/tags.py]: Make sure you run black before requesting a review.

**Review by mudkipdev — CHANGES_REQUESTED** [review]

File: leaf/extensions/tags.py (hunk 2 above)
Comment (bsrecker) [reply leaf/extensions/tags.py]: Done

**Comment by mudkipdev** [comment]
Looks good to me, merging now.

**Pull request merged**